src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java

```diff
@@ -184,8 +186,19 @@ private RequestQueue setHoldShelfExpirationDateWithExpirationDateManagement(
 
     ExpirationDateManagement expirationDateManagement = calculatedRequest.getPickupServicePoint()
       .getHoldShelfClosedLibraryDateManagement();
-    String intervalId = calculatedRequest.getPickupServicePoint().getHoldShelfExpiryPeriod()
-      .getIntervalId().toUpperCase();
+
+    String intervalId;
+    TimePeriod holdShelfExpiryPeriod = calculatedRequest.getPickupServicePoint()
+      .getHoldShelfExpiryPeriod();
+    if (holdShelfExpiryPeriod != null && isNotBlank(holdShelfExpiryPeriod.getIntervalId())) {
```

Comment:
You are calling `holdShelfExpiryPeriod.getIntervalId()` three times here. How about an `Optional`?

```
    String intervalId = Optional.of(calculatedRequest)
      .map(Request::getPickupServicePoint)
      .map(ServicePoint::getHoldShelfExpiryPeriod)
      .map(TimePeriod::getIntervalId)
      .map(String::toUpperCase)
      .orElse(NOT_DEFINED_INTERVAL);

    log.info("setHoldShelfExpirationDateWithExpirationDateManagement:: interval: {}", intervalId);
```

Reply:
done